### PR TITLE
Artists Repayment

### PR DESCRIPTION
**Pull request by sachasmart-weavik**
### Description

> Allows the artist to repay their loans. Also added some functionality so new devs can come in fresh and not have to worry about contracts. 

[FAD-171]
### Checklist

- [ ] `CHANGELOG` has been updated (_if appropriate_)
- [ ] Environment variables updated (_if appropriate_)

[FAD-171]: https://weavik.atlassian.net/browse/FAD-171?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/client3/constants/index.ts [hunk 1]
```diff
@@ -28,10 +28,13 @@ export const API_BASE_URL = process.env.NEXT_PUBLIC_GCLOUD_FUNCTIONS_URL
   ? process.env.NEXT_PUBLIC_GCLOUD_FUNCTIONS_URL
   : networkName === "mainnet"
   ? "https://us-central1-free-artists.cloudfunctions.net"
-  : networkName === "murmuration"
-  ? "https://murmuration.goldfinch.finance/_gcloudfunctions"
+  : networkName === "aurora_testnet"
+  ? "https://us-central1-free-artists.cloudfunctions.net"
+  : networkName === "localhost"
+  ? "http://localhost:5001/goldfinch-frontends-dev/us-central1"
   : "http://localhost:5001/goldfinch-frontends-dev/us-central1";
```
Comment (v1n33th) [inline packages/client3/constants/index.ts:35]: Very minor thing, I think there is no need to check for networkName as localhost, could just default to localhost url
```suggestion
  : "http://localhost:5001/goldfinch-frontends-dev/us-central1";
```

Comment (sachasmart-weavik) [reply packages/client3/constants/index.ts:35]: Yeap. Fair point. Thanks!

File: packages/client3/constants/index.ts [hunk 2]
```diff
@@ -57,13 +60,14 @@ export const PERSONA_CONFIG: PersonaConfig =
...
     ? ""
-    : networkName === "murmuration"
-    ? "https://murmuration.goldfinch.finance"
-    : "https://us-central1-free-artists-dev.cloudfunctions.net";
-// : "http://localhost:4000";
+    : networkName === "aurora_testnet"
+    ? "https://us-central1-free-artists.cloudfunctions.net"
+    : networkName === "localhost"
+    ? "http://localhost:4000"
+    : "http://localhost:4000";
```
Comment (v1n33th) [inline packages/client3/constants/index.ts:65]: Same as above.
```suggestion
        : "http://localhost:4000";
```

File: packages/client3/pages/artist/pool/[address]/artist-pool-information.tsx [hunk 3]
```diff
@@ -88,56 +93,44 @@ function ArtistWithdrawPool({
               {numOfBackers > 1 ? "Backers" : "Backer"}
             </BodyText>
           </div>
+          {!lockedPool ? (
+            <div className="mt-8 w-full border border-dark-90" />
+          ) : (
+            <></>
```
Comment (v1n33th) [inline packages/client3/pages/artist/pool/[address]/artist-pool-information.tsx:99]: You can also return null instead of React fragment(<></>)

Comment (v1n33th) [reply packages/client3/pages/artist/pool/[address]/artist-pool-information.tsx:99]:
```suggestion
            null
```

Comment (sachasmart-weavik) [reply packages/client3/pages/artist/pool/[address]/artist-pool-information.tsx:99]: Ahh thank you!!

File: packages/client3/pages/artist/pool/[address]/artist-pool-information.tsx [hunk 4]
```diff
@@ -88,56 +93,44 @@ function ArtistWithdrawPool({
...
       </div>
+      {lockedPool ? (
+        <div className="mt-8">
+          {console.log(
+            "balance",
+            BigNumber.from(balance).toNumber(),
+            lockedPool,
+            balance,
+            deposited
+          )}
```
Comment (v1n33th) [inline packages/client3/pages/artist/pool/[address]/artist-pool-information.tsx:121]: 🤔

File: packages/client3/pages/artist/pool/[address]/artist-pool-information.tsx [hunk 5]
```diff
@@ -88,56 +93,44 @@ function ArtistWithdrawPool({
...
+          <ArtistRepayCard
+            balance={balance}
+            onSubmit={(amount) => {
+              onRepayment?.(amount);
+            }}
+          />
+        </div>
+      ) : (
+        <></>
```
Comment (sachasmart-weavik) [inline packages/client3/pages/artist/pool/[address]/artist-pool-information.tsx:130]: null

File: packages/client3/pages/artist/pool/[address]/artist-pool-information.tsx (hunk 3 above)
Comment (v1n33th) [reply packages/client3/pages/artist/pool/[address]/artist-pool-information.tsx:99]: Could be added to other such returns as well

File: packages/client3/pages/artist/pool/[address]/index.page.tsx [hunk 6]
```diff
@@ -88,16 +88,19 @@ function ArtistPoolPage() {
     },
   });
 
+  console.log(poolMetaData?.poolAddress);
```
Comment (v1n33th) [inline packages/client3/pages/artist/pool/[address]/index.page.tsx:91]: Could be removed

File: packages/client3/pages/artist/pool/[address]/index.page.tsx [hunk 7]
```diff
@@ -88,16 +88,19 @@ function ArtistPoolPage() {
...
+
   const { data: { tranchedPool: tranchedPoolData } = {} } =
     useArtistPoolGraphDataQuery({
       skip: !poolMetaData?.poolAddress,
       variables: {
-        tranchedPoolAddress: poolMetaData?.poolAddress || "",
+        tranchedPoolAddress: "0xd010269dca2f8f958f3a498ce0fecc324c594e7d" || "",
```
Comment (v1n33th) [inline packages/client3/pages/artist/pool/[address]/index.page.tsx:97]: Not intended I believe

File: packages/client3/pages/artist/pool/[address]/pool-details-grid-right.tsx [hunk 8]
```diff
@@ -18,27 +17,27 @@ type Props = {
...
 
-  // Note: Need to use Borrower.sol to interact with TranchedPool.sol
   const borrowerContract = useContract("Borrower", poolData.borrowerContract);
 
-  const goldfinchFactory = useContract(
-    "GoldfinchFactory",
-    CONTRACT_ADDRESSES.GoldfinchFactory
+  const USDC = useContract(
+    "USDC",
+    "0x3E0B09aDf6171F5D1aefef567BA6Cf1fb364E080"
```
Comment (v1n33th) [inline packages/client3/pages/artist/pool/[address]/pool-details-grid-right.tsx:24]: Can this address be moved to CONTRACT_ADDRESSES?

Comment (sachasmart-weavik) [reply packages/client3/pages/artist/pool/[address]/pool-details-grid-right.tsx:24]: Yeap you're right.

File: packages/client3/pages/artist/pool/[address]/pool-details-grid-right.tsx [hunk 9]
```diff
@@ -18,27 +17,27 @@ type Props = {
...
   );
 
-  const disabled = !(
-    tranchedPoolData.estimatedTotalAssets - tranchedPoolData.totalDeployed
+  const CreditLine = useContract(
+    "CreditLine",
+    "0x099758bd8b67658e153357d302da655c52b3aefd"
```
Comment (v1n33th) [inline packages/client3/pages/artist/pool/[address]/pool-details-grid-right.tsx:29]: Can this be a constant as well?

Comment (sachasmart-weavik) [reply packages/client3/pages/artist/pool/[address]/pool-details-grid-right.tsx:29]: Yeah I just saw this too. Nah I need to change it to the specific pool's creditline.

Comment (v1n33th) [reply packages/client3/pages/artist/pool/[address]/pool-details-grid-right.tsx:29]: Also variable could be camelCased

**Review by v1n33th — COMMENTED** [review]
Awesome..Added some suggestions

**Comment by sachasmart-weavik** [comment]
> Awesome..Added some suggestions

Ah my bad man. Thank you!!

**Review by v1n33th — APPROVED** [review]

**Pull request merged**